src/api/pipelineRuns.test.js

```diff
@@ -326,3 +328,226 @@ it('startPipelineRun', () => {
+        'kubectl.kubernetes.io/last-applied-configuration':
+          '{"apiVersion": "tekton.dev/v1beta1", "keya": "valuea"}'
+      },
+      labels: {
+        keyl: 'valuel',
+        key2: 'value2',
+        'tekton.dev/pipeline': 'foo'
+      },
+      uid: '111-233-33',
+      resourceVersion: 'aaaa'
+    },
+    spec: {
+      pipelineRef: {
+        name: 'simple'
+      },
+      params: [{ name: 'param-1' }, { name: 'param-2' }]
+    },
+    status: { startTime: '0' }
+  };
+  const { namespace, payload } = generateNewPipelineRunPayload(pr, true);
+  const expected = `apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  annotations:
+    keya: valuea
+  generateName: test-r-
+  labels:
+    keyl: valuel
+    key2: value2
+    dashboard.tekton.dev/rerunOf: test
+  namespace: test-namespace
+spec:
+  pipelineRef:
+    name: simple
+  params:
+    - name: param-1
+    - name: param-2
+`;
+  expect(namespace).toEqual('test-namespace');
+  expect(yaml.dump(payload)).toEqual(expected);
+});
+
+it('generate new pipeline run minimum', () => {
+  const pr = {
+    apiVersion: 'tekton.dev/v1beta1',
+    kind: 'PipelineRun',
+    metadata: {
+      name: 'test',
+      namespace: 'test-namespace'
+    },
+    spec: {
+      pipelineRef: {
+        name: 'simple'
+      }
+    }
+  };
+  const { namespace, payload } = generateNewPipelineRunPayload(pr, false);
+  const expected = `apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: test-
+  namespace: test-namespace
+spec:
+  pipelineRef:
+    name: simple
+`;
+  expect(namespace).toEqual('test-namespace');
+  expect(yaml.dump(payload)).toEqual(expected);
+});
+it('generate new pipeline run maximum', () => {
+  const pr = {
+    apiVersion: 'tekton.dev/v1beta1',
+    kind: 'PipelineRun',
+    metadata: {
+      annotations: {
+        keya: 'valuea',
+        'kubectl.kubernetes.io/last-applied-configuration':
+          '{"apiVersion": "tekton.dev/v1beta1", "keya": "valuea"}'
+      },
+      labels: {
+        keyl: 'valuel',
+        key2: 'value2',
+        'tekton.dev/pipeline': 'foo'
+      },
+      name: 'test',
+      namespace: 'test-namespace',
+      uid: '111-233-33',
+      resourceVersion: 'aaaa'
+    },
+    spec: {
+      pipelineRef: {
+        name: 'simple'
+      },
+      params: [{ name: 'param-1' }, { name: 'param-2' }]
+    },
+    status: { startTime: '0' }
+  };
+  const { namespace, payload } = generateNewPipelineRunPayload(pr, false);
+  const expected = `apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  annotations:
+    keya: valuea
+  generateName: test-
+  labels:
+    keyl: valuel
+    key2: value2
+  namespace: test-namespace
+spec:
+  pipelineRef:
+    name: simple
+  params:
+    - name: param-1
+    - name: param-2
+`;
+  expect(namespace).toEqual('test-namespace');
+  expect(yaml.dump(payload)).toEqual(expected);
+});
+it('generate new pipeline run last applied configuration should be removed', () => {
```

Comment:
This could be combined with one of the tests above to avoid all the setup duplication, same for the labels test below. Logically it's part of the same processing, preserving only the relevant annotations/labels on the generated payload.